### PR TITLE
Sass 1032

### PR DESCRIPTION
**Pull request by michael-rob-poole**
### Description
Creation of Employer Pay Amount page

Add a link to the relevant story in Jira
[SASS-859](https://jira.tools.tax.service.gov.uk/browse/SASS-859)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of master?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current master that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of master?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/forms/validation/mappings/Formatters.scala [hunk 1]
```diff
@@ -39,9 +39,9 @@ trait Formatters {
 
   private[mappings] def currencyFormatter(requiredKey: String,
                                           invalidNumericKey: String,
-                                          nonNumericKey: String,
                                           maxAmountKey: String,
-                                          args: Seq[String] = Seq.empty): Formatter[BigDecimal] =
+                                          args: Seq[String] = Seq.empty[String]
```
Comment (BenCosfordHMRC) [inline app/forms/validation/mappings/Formatters.scala:43]: is nonNumericKey not used anywhere then?

File: app/models/employment/EmploymentDetailsView.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                                 startDate: Option[String],
+                                 cessationDateQuestion: Option[Boolean],
+                                 cessationDate: Option[String],
+                                 taxablePayToDate: Option[BigDecimal],
+                                 totalTaxToDate: Option[BigDecimal],
+                                 tipsAndOtherPaymentsQuestion: Option[Boolean],
+                                 tipsAndOtherPayments: Option[BigDecimal],
+                                 isUsingCustomerData: Boolean)
+
+object EmploymentDetailsView {
```
Comment (BenCosfordHMRC) [inline app/models/employment/EmploymentDetailsView.scala:33]: this file can go I think, think its a duplicate

File: it/controllers/employment/EmployerPayAmountControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+        val inYearUrl =s"$appUrl/$taxYear/how-much-pay?employmentId=001"
+        urlGet(inYearUrl, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+      }
+
+
+      "has an SEE_OTHER status" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some("http://localhost:11111/income-through-software/return/2022/view")
+      }
+    }
```
Comment (BenCosfordHMRC) [inline it/controllers/employment/EmployerPayAmountControllerISpec.scala:249]: these show tests could be moved up to the prior .show tests and utilise the us.isAgent to cover more scenarios when they're an agent

File: it/controllers/employment/EmployerPayAmountControllerISpec.scala [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+          inputFieldValueCheck("9999999999999999999999999999", inputAmountField)
+          errorSummaryCheck(maxAmountErrorText, expectedErrorHref)
+        }
+      }
+    }
+  }
+
+  ".submit" should {
+
+    "redirect to Overview page when a valid form is submitted" when {
```
Comment (BenCosfordHMRC) [inline it/controllers/employment/EmployerPayAmountControllerISpec.scala:327]: as above this test could be moved up to the prior .submit tests and utilise the us.isAgent to cover more scenarios when they're an agent

**Review by BenCosfordHMRC — CHANGES_REQUESTED** [review]

File: app/forms/validation/mappings/Formatters.scala (hunk 1 above)
Comment (michael-rob-poole) [reply app/forms/validation/mappings/Formatters.scala:43]: Not here.  I think it used to be used for the message Enter numbers 0-9 etc. Like below in personal

Comment (michael-rob-poole) [reply app/forms/validation/mappings/Formatters.scala:43]: `
  def amountForm(
                  emptyFieldKey: String,
                  wrongFormatKey: String = "common.error.invalid_currency_format",
                  exceedsMaxAmountKey: String = "common.error.amountMaxLimit",
                  emptyFieldArguments: Seq[String] = Seq.empty[String]
                ): Form[BigDecimal] = Form(
    amount -> currency(
      requiredKey = emptyFieldKey,
      invalidNumeric = wrongFormatKey,
      nonNumericKey = wrongFormatKey,
      maxAmountKey = exceedsMaxAmountKey,
      args = emptyFieldArguments
    )
  )`

File: app/models/employment/EmploymentDetailsView.scala (hunk 2 above)
Comment (michael-rob-poole) [reply app/models/employment/EmploymentDetailsView.scala]: Done

**Review by BenCosfordHMRC — APPROVED** [review]

**Pull request merged**